llvm-project/stage2.cmake

```diff
@@ -1,22 +1,74 @@
-# TODO: comment every one of these.
+# Please try to keep these cmake variables alphabetically sorted.
+
+# Enable optimizations, as opposed to a debug build.
 set(CMAKE_BUILD_TYPE "Release" CACHE STRING "")
+
+# Use stage1's clang to bootstrap stage2, as opposed to Alpine's clang (which
+# shouldn't be installed on the system at this point, but would be installed to
+# /usr/bin/clang).
 set(CMAKE_CXX_COMPILER "/usr/local/bin/clang++" CACHE FILEPATH "")
+
+# Use libc++ from stage1.
+# We could not do this in stage1 because Alpine doesn't distribue libc++ ATM.
+# TODO: does clang cmake not have a CLANG_DEFAULT_* flag for setting this?
 set(CMAKE_CXX_FLAGS "--stdlib=libc++" CACHE STRING "")
+
+# See above comment for CMAKE_CXX_COMPILER.
 set(CMAKE_C_COMPILER "/usr/local/bin/clang" CACHE FILEPATH "")
+
+# Use libunwind from stage1.
+# Statically link resulting executable.
+# TODO: why is -lc++abi necessary for static link?
```

Comment:
Because static libraries do not carry `DT_NEEDED` and thus we cannot get the transitive closure for the symbols and will fail to link because `lld` behaves incorrectly (undefined symbols are encouraged by ELF).

Reply:
> because lld behaves incorrectly (undefined symbols are encouraged by ELF).

"Behaves incorrectly?"  Does BFD do something differently/correctly?

Reply:
Yes, BFD does permit the undefined symbols IIRC.  However, re-reading the lines below, the executable should be fully linked and not have undefined symbols - that does make sense (even though it may be different from the default expected semantics).

That said, I think this is more about the expectations and Unix defaults ... its just different and correctness here refers to the fact that it changes the expectations by default - which means that the behaviour is now different which for a drop-in replacement is arguably incorrect.

Reply:
cc @MaskRay